WPB-25915 add timeout and duration metric for conversation migration (#5244)
wireapp/wire-server: docs/src/developer/reference/config-options.md

channels from the team-management UI. It is highly recommended to take a backup
of the affected Cassandra data before triggering a migration.

When migrating conversations, `background-worker.config.migrateConversationsOptions.timeout`
should be configured as well. It sets a per-conversation upper bound for the
migration attempt, so a stuck conversation does not keep the migration run
blocked indefinitely. Start with a value that is comfortably above the normal
time for one conversation migration, then adjust it based on observed runtime

> and the size of your dataset.

Reviewer: This might not be a good advise, stopping the migration and restarting it can cause it to fail due to tombstones. Can we perhaps find a sensible default?

migrateTeamFeatures: false
migrateDomainRegistration: false

# conversation migration settings
migrateConversationsOptions:
  pageSize: 10000
  parallelism: 2
  # (optional) migration timeout in seconds, applies to a single conversation

>   timeout: 60

Reviewer: ```suggestion
  # (optional) migration timeout, applies to a single conversation
  timeout: 5s
```

I'd recommend using the  `Duration` type, and then we should just expect people to specify units instead. 
I also think 60s might be too much to suggest, let's make it 5s?

federationDomain: example.org
```

The optional `migrateConversationsOptions.timeout` setting limits how long a single
conversation migration attempt may run after it has acquired the migration
lock. The value is a plain number of seconds, so `60` means 1 minute.
If the timeout is exceeded, that conversation migration is aborted and the

> whole migration run is treated as failed.

Reviewer: ```suggestion
The optional `migrateConversationsOptions.timeout` setting limits how long a single
conversation migration attempt may run after it has acquired the migration
lock. If the timeout is exceeded, that conversation migration is aborted and the
migration of this conversation is treated as failed.
```

Choose a value that is comfortably above the normal time for one conversation
migration, but still low enough to catch a genuinely stuck migration in a
reasonable time. A good starting point for most deployments is `60` seconds

> (1 minute), then adjust based on observed migration durations.

Reviewer: ```suggestion
reasonable time. A good starting point for most deployments is `5s`, 
then adjust based on observed migration durations.
```

If the setting is omitted, no timeout will be enforced. If a conversation

> migration stalls, this can lead to leaked exclusive advisory locks.

Reviewer: Makes me wonder if we should really have an operating mode this dangerous. 🤔
